### PR TITLE
grcov parsing of gcda files 

### PR DESCRIPTION
**Pull request by rhcu**
Issue #2 
I have made parsing of .gcda files stored in the temporary GCOV directory to store coverage information in `coveralls.json` file with grcov. Used subprocesses to call the shell command in the script and to output the result to `coveralls.json`. Shutil module was used to zip .gcda files.
I am not sure how to check whether the generated report (coveralls.json) is right. Can you suggest a way? I have added it to the commit too.

### CODE REVIEW

**Comment by marco-c** [comment]
The quickest way is to parse the JSON file with Python and check that some function that is expected to be covered is actually covered.
For example, you can check if https://dxr.mozilla.org/mozilla-central/source/toolkit/xre/nsAppRunner.cpp#3125 is covered (this function is executed at startup in Firefox, so it has to be covered).

File: crawler.py [hunk 1]
```diff
@@ -191,10 +193,25 @@ def run_all(driver):
...
+
     # Webdriver uses Firefox Binaries from downloaded cov build
     driver = webdriver.Firefox(firefox_binary='tools/firefox/firefox-bin')
     run_all(driver)
+
+    # Zip gcda file from gcov directory
+    shutil.make_archive('code-coverage-gcda', 'zip', gcov_dir)
+    grcov_command = [
+        os.path.join('tools', 'grcov'),
+        '-t', 'coveralls',
```
Comment (marco-c) [inline crawler.py]: Let's use `coveralls+`, whose output also contains functions.

File: crawler.py [hunk 2]
```diff
@@ -191,10 +193,25 @@ def run_all(driver):
...
+        '-p', prefix,
+        os.path.join('tools', 'target.code-coverage-gcno.zip'), 'code-coverage-gcda.zip',
+        '--token', 'UNUSED',
+        '--commit-sha', 'UNUSED'
+    ]
+    with open('coveralls.json', "w+") as outfile:
+        subprocess.call(grcov_command, stdout=outfile)
```
Comment (marco-c) [inline crawler.py]: Let's use `check_call` instead of `call`, so the function will raise an exception if grcov fails.

**Comment by marco-c** [comment]
Once you've verified that the coveralls.json file is looking good, please remove it from the PR.

File: crawler.py [hunk 3]
```diff
@@ -191,10 +193,25 @@ def run_all(driver):
...
+    shutil.make_archive('code-coverage-gcda', 'zip', gcov_dir)
+    grcov_command = [
+        os.path.join('tools', 'grcov'),
+        '-t', 'coveralls',
+        '-p', prefix,
+        os.path.join('tools', 'target.code-coverage-gcno.zip'), 'code-coverage-gcda.zip',
+        '--token', 'UNUSED',
+        '--commit-sha', 'UNUSED'
+    ]
+    with open('coveralls.json', "w+") as outfile:
```
Comment (marco-c) [inline crawler.py]: Let's call this `output.json`

**Comment by rhcu** [comment]
Ok. Will verify the file with Python script. Thanks for comments. :)

**Comment by rhcu** [comment]
Now, there are also functions in the JSON file and deleted this file from the PR.  Also, added a line to delete zipped gcda files from the directory after parsing.

**Pull request merged**